samtranslator/model/api/api_generator.py

```diff
@@ -184,8 +185,12 @@ def _construct_stage(self, deployment, swagger):
         # If StageName is some intrinsic function, then don't prefix the Stage's logical ID
         # This will NOT create duplicates because we allow only ONE stage per API resource
         stage_name_prefix = self.stage_name if isinstance(self.stage_name, string_types) else ""
-
-        stage = ApiGatewayStage(self.logical_id + stage_name_prefix + 'Stage',
+        if '-' in stage_name_prefix or '_' in stage_name_prefix:
```

Comment:
According to the [CloudFormation docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resources-section-structure.html#resources-section-structure-logicalid), the only valid characters are alphanumeric (A-Za-z0-9). Why don't you change this check to be for any non-alphanumeric characters rather than just `-` or `_`.

Reply:
I just changed the condition in the latest commit.